src/driver/shader.rs

```diff
@@ -483,6 +489,11 @@ impl Shader {
         )
     }
 
+    /// Returns the reflected entry point of this shader.
+    pub fn entry_point(&self) -> &EntryPoint {
```

Comment:
I made this item available with a non-mutable function because it was on the `Shader` type as a cached item, and I don't want you to be able to swap it out with another value - because I already have the source of truth which is the shader code and the entry point name.

Likewise I kept the `entry_point` field private so that a user is forced to go through the "builder" path to make an instance and therefore they can't get around the reflection. This is very pedantic but I'm trying to limit the possibility of using the API to do something that causes the rest of the code to not function as described.